test/extended/federation.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Runs the federation tests against a standard started server.
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+source "${OS_ROOT}/test/extended/setup.sh"
+
+os::test::extended::setup
+os::util::ensure::built_binary_exists kubefed
+os::test::extended::focus "$@"
+
+export OS_FEDERATION_NAME="${OS_FEDERATION_NAME:-origin-federation}"
+export OS_FEDERATION_NAMESPACE="${OS_FEDERATION_NAMESPACE:-federation-system}"
+export OS_FEDERATION_SERVICE_ACCOUNT="system:serviceaccount:${OS_FEDERATION_NAMESPACE}:default"
+
+os::test::junit::declare_suite_start "extended/federation"
+
+os::log::info "Deploying federation control plane"
+os::cmd::expect_success "kubefed init '${OS_FEDERATION_NAME}' --dns-provider=google-clouddns \
```

Comment:
As per our discussion in standup, ``oc cluster up`` can be used with ``--etcd-persistent-storage=true`` which ensures ``/var/etc/data`` will be writeable.